### DISCUSSION TITLE
Localstorage persistence doesn't set initial value

### DISCUSSION THREAD

**Discussion by Ranguna** [arnelenero/simpler-state#22] [General]
**Describe the bug**
Localstorage persistence doesn't persist initial values.

**To Reproduce**
1. Go to https://codesandbox.io/s/pedantic-brook-t5lve?file=/src/App.tsx
2. Check localstorage: there's no value for "todo".
3. Uncomment line 7 on the App.tsx file
4. Recheck localstorage and you should see a value set for "todo".

**Expected behavior**
On step 2, I was expecting to see the value for "todo" set to the initial value that was provided. 
In other words: I was expecting the lib to automatically save the initial value in localStorage when there is no saved value. 

**Desktop (please complete the following information):**
 - OS: linux
 - Browser firefox
 - Version 92

**Additional context**
A scenario where an entity value might be persisted before being explicitly set with the `set` method might be kinda weird, but it's a use case for me when generating random values for a request. A random value is only set once (as an initial value) and it's reused afterwards without having to be re-set again.

**Comment by arnelenero** [comment]
Thank you for your submission.

I'm not sure I see where the error is.  I forked your sandbox and am able to display the initial value when there's no saved value in localStorage:  https://codesandbox.io/s/nice-shtern-jlq0z?file=/src/App.tsx

However, I think what you're looking for is for it to automatically save the initial value in localStorage when there is no saved value?

EDIT: After reading your Expected Behavior one more time, I think my understanding above would be correct. 😀

I asked a few users in the past about this behavior, and majority voted for the current behavior.  But I personally think your expected behavior makes perfect sense.  Let me rethink this again 😂.  

For now the immediate workaround would be (similar to what you did in your sandbox):
1.  Set the initial state to something like empty string.
2.  In a useEffect, check if entity value is the same as above initial value, and if and only if so, set value to your actual desired initial value.

**Comment by Ranguna** [comment]
@arnelenero oh damn, that's what happens when your write issues at 2 am xD
Sorry about that, I'll rewrite the description. But yeah, that's exactly what I was expecting.

Take your time 👌
And feel free to close this issue if you think the current behaviour is better.

Awesome lib btw and thank you very much for taking a look at this!
